src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -346,42 +347,20 @@ public void stopGame() {
     }
   }
 
-  private void autoSave() {
+  private void autoSave(final String fileName) {
     SaveGameFileChooser.ensureMapsFolderExists();
-    final File f1 =
-        new File(ClientContext.folderSettings().getSaveGamePath(), SaveGameFileChooser.getAutoSaveFileName());
-    final File f2 =
-        new File(ClientContext.folderSettings().getSaveGamePath(), SaveGameFileChooser.getAutoSave2FileName());
-    final File f;
-    if (f1.lastModified() > f2.lastModified()) {
-      f = f2;
-    } else {
-      f = f1;
-    }
-
-    try (FileOutputStream out = new FileOutputStream(f)) {
-      saveGame(out);
-    } catch (final Exception e) {
-      ClientLogger.logQuietly(e);
-    }
+    saveGame(new File(ClientContext.folderSettings().getSaveGamePath(), fileName));
   }
 
-  private void autoSaveRound() {
-    SaveGameFileChooser.ensureMapsFolderExists();
-    final File autosaveFile;
-    if (m_data.getSequence().getRound() % 2 == 0) {
-      autosaveFile =
-          new File(ClientContext.folderSettings().getSaveGamePath(), SaveGameFileChooser.getAutoSaveEvenFileName());
-    } else {
-      autosaveFile =
-          new File(ClientContext.folderSettings().getSaveGamePath(), SaveGameFileChooser.getAutoSaveOddFileName());
-    }
+  private void autoSaveAfter(final IDelegate currentDelegate) {
+    final String phaseName = currentDelegate.getClass().getTypeName().substring(
```

Comment:
Referencing package names and sub classes directly by name in strings can be quite brittle. If we rename or move package, this code would silently break. It also does not seem appropriate for a ServerGame to know so many details about a delegate and getting the name of the current phase (law of demeter violation)

I suggest to fix these problems by adding a `getPhaseName()` method to `IDelegate`. Then testing this code would also be very easy to do as well.